### ISSUE TITLE
[performance issue] long time rendering of MultiComboBox

### ISSUE THREAD

**Issue by natsu0728** [SAP/ui5-webcomponents#3614]
### **Bug Description**
When I have more than 100 options of MultiComboBox, I easily get a dropped frame which has 0 Fps. 
I used Chrome Performance to record and it gives the possible reason.
![image](https://user-images.githubusercontent.com/22427546/128673285-aa5d4964-e06e-484f-99cc-4972e5235fd7.png)

getComputedStyle in getEffectiveDir.js will cause forced reflow
![image](https://user-images.githubusercontent.com/22427546/128673355-a1e07d2e-ecd9-4764-8b18-151d56b25e0a.png)

### **Isolated Example**
[https://codesandbox.io/s/reverent-glade-h92rj?file=/src/App.js](https://codesandbox.io/s/reverent-glade-h92rj?file=/src/App.js).
The page is blocked for about 5 seconds after clicking "show multicombobox"

### **Expected Behavior**
avoid performance issue caused by forced reflow

### **Context**
- UI5 Web Components version: 1.0.0-rc.14
- OS/Platform: mac
- Browser: chrome
- Affected component: MultiComboBox

### **Priority**
- [ ] Low
- [x] Medium
- [ ] High
- [ ] Very High

**Comment by pskelin** [comment]
Can you please share an example to reproduce, or at least specific number of multi combox instances and how many options?

the multicombox is designed to be used for up to 200 elements, we should upgrade the docs, but here is the text from openui5:
```
Usage
When to use:
- The user needs to select one or more options from a long list of options (maximum of approximately 200).
When not to use:
- The user needs to choose between two options such as ON or OFF and YES or NO. In this case, consider using a switch control instead
- You need to display more that one attribute. In this case, consider using the select dialog or value help dialog instead.
- The user needs to search on multiple attributes. In this case, consider using the select dialog or value help dialog instead.
- Your use case requires all available options to be displayed right away, without any user interaction. In this case, consider using the checkboxes instead.
```
https://sapui5.hana.ondemand.com/#/api/sap.m.MultiComboBox

Please also note how Chrome states this is *likely* a performance issue. Such kinds of warnings are only a problem if the browser calculates the layout once and it is thrown away to be calculated again afterwards. This is not the case usually, if we remove the forced style calculation, the browser still has to calculate the layout at the end. This means the scenario will remain the same, just the warning will be gone.

Let us know with a specific example or at least exact number of controls you used in your example before proceeding any further.

**Comment by natsu0728** [comment]
Hi, pskelin
Thanks for answering. I mocked my case in codeSandbox, see it here: [https://codesandbox.io/s/reverent-glade-h92rj?file=/src/App.js](https://codesandbox.io/s/reverent-glade-h92rj?file=/src/App.js)

I have 5 multicomboboxes in this filterbar, and each one has 200 options. When you click "show multiComboBox" button, you have to wait 5 to 6 seconds until the page finishes rendering these components.
In my app, the behaviour is that I have to wait the same long time when going to this page through router change.

Also I've tried to remove the forced reflow in getEffectiveDir.js and the rendering time reduced to 1 second, unfortunately I don't have a way to show this comparison in codeSandbox.

**Comment by pskelin** [comment]
Hi @natsu0728 ,
thanks for the great sample and analysis. This is really a problem we had not thought about and are going to address.

There are two options we are checking. One is to render the multi combo box popover only after it is clicked (which will be more of a quick fix and a good optimisation on its own).

The second one would require a bigger change in CSS to get rid of the getComputedStyle you identified and switch to CSS Logical Properties for right-to-left (RTL) text support (this is the reason for the getComputedStyle call).

You can track the progress here in case there are follow-up issues.

Thanks again for the nice report.

**Comment by ChristyFei** [comment]
How about the latest progress? Thank you.

**Comment by ilhan007** [comment]
Hello @natsu0728 @ChristyFei with the latest development the rendering of components implementing right-to-left (RTL)  supports has been improved. Previously we had many getComputedStyle calls to detect the page direction (rtl, ltr) to apply some styles. Now, we switched to CSS Logical Properties for right-to-left (RTL) text support and let the browser do it for us, reduced the calls to getComputedStyle to minimum and improved the performance significantly.

The rendering time for your use-case with multiple MultiCombox components with many items  has been reduced from 5 sec to 1 sec. You can check it here: 
https://codesandbox.io/s/dreamy-tharp-5ffxwy?file=/package.json

Notice the `@ui5/webcomponents": "0.0.0-3236d56be` version in the package.json. The `0.0.0-3236d56be` is a preview version pointing to the very latest development - our master branch. Before the end of the month, the improvement will be released with the next minor  version - 1.4.0
